airflow/providers/http/hooks/http.py

```diff
@@ -102,6 +113,10 @@ def get_conn(self, headers: dict[Any, Any] | None = None) -> requests.Session:
                 self.base_url = self.base_url + ":" + str(conn.port)
             if conn.login:
                 session.auth = self.auth_type(conn.login, conn.password)
+            elif self._auth_type and self.auth_args:
+                    session.auth = self.auth_type(*self.auth_args)
+            elif self._auth_type:
+                session.auth = self.auth_type()
```

Comment:
I feel it’s good enough to remove `self.auth_args` and simply do `self.auth_type()` here. The user can always use `functools.partial()` if arguments are needed; I think this fits how Python generally design these sort of things. (We should mention this in the docstring, of course.)

Reply:
Fair, this way we avoid adding more parameters to an already long list. Will follow up with update soon.

What do you think about need of changing other tests because we change the default `auth_type`? Should I proceed with adjusting them?

Reply:
Yeah changing the tests feels appropriate. (Maybe change a few first to see if anyone has issues with the way things are.)

Reply:
Hmm, it looks like it wasn't the tests but the `livy` hook which needed a change, please see the comment and let me know if it is ok to adjust it like that.